twine/repository.py

```diff
@@ -121,9 +121,14 @@ def _upload(self, package):
                 (package.basefilename, fp, "application/octet-stream"),
             ))
             encoder = MultipartEncoder(data_to_send)
-            bar = ProgressBar(expected_size=encoder.len, filled_char='=')
+            bar = ProgressBar(total=encoder.len, unit='bytes', unit_scale=True, leave=False)
+            def update_progressbar(monitor, _cache=[0]):
```

Comment:
Why not use a variable that lives at the same scope as the function instead of relying on this. I agree with @SethMichaelLarson that this isn't going to be easy to maintain.